packages/wells/src/client/api/wells.ts

```diff
@@ -236,4 +250,71 @@ export class Wells extends AssetsAPI {
 
     return await this.getByIds(assetIds);
   };
+
+  public filterPolygonSearch = async ({
+    geometry,
+    crs = 'epsg:4326',
+    outputCrs = 'EPSG:4326',
+    limit = 1000,
+    name,
+    id,
+    data_source,
+    operator,
+    field,
+    block,
+    customFilter = undefined,
+  }: {
+    geometry: string | GeoJson;
+    crs?: string;
+    outputCrs?: string;
+    limit?: number;
+    name?: string;
+    id?: number;
+    data_source?: string[];
+    operator?: string[];
+    field?: string[];
+    block?: string[];
```

Comment:
how do you propose adding my next filters, for example i need:



- GammaRay
- Caliper
- Resistivity 
- Density
- Neutron
- PPFG
- Geomechanics
- Core
- NDS - (Non Drilling Surprise)
  - Risk Type
  - Severity
  - Probability
- NPT - (Non Productive Time)
  - Duration
  - Detail
  - Code
- KB elevation
- TVD
- Water Depth
- Spud date
- Deviation Cutoff Inclination Angle
- Dog Leg Severity

perhaps a FilterQuery that we can extend from?

eg: https://github.com/cognitedata/cognite-sdk-js/blob/2c0356bccef4f38b0365448f7eb66230ef6c9e53/packages/stable/src/types.ts#L1288

Reply:
So you mean the input to the method would just be a single object that has attributes which contain all the filters you want? I don't think we can use an extension on FilterQuery as we aren't doing the filtering in the call to CDF, but we could create a similar type which contains all the filters you need (which might be what you meant in the first place)

Reply:
Most of these other filters fall into the "measurement" category, which imho should be treated separately. 

So, while the ones implemented in this PR (and what we plan to treat as simple labels and/or metadata) are reasonably basic string attribute filters that seem like a good choice to live as first-class citizens of the list method, I guess these other ones you mentioned would implement special filters. 

So we planned to have a hasMeasurements(measurements=[some list of measurement names])  - implemented by relationships - and a measurementValues, and, for this second  case, indeed we would need a MeasurementFilter class that has to be specialized or extended by each special class of measurement where we want to filter values from (because some of them will be special little snowflakes that will need some sort of rules to filter)

Reply:
But, again, now I look at it and think this would be the most idiomatic Python way of doing stuff, and I'm out of my league trying to use it in the JS SDK :) - so, if the idiomatic JS way of doing it would be having an interface FilterQuery for allfilters, by all means, let's do it!

Reply:
The newest commit should address these concerns @rek I've created a WellFilter interface which can be extended in the future to include all of these attributes, without breaking any existing code, let me know what you think

Reply:
excellent. was mainly showing that list to point out that fact we needed the interface more generic, and now i see you have done exactly that! 💯 🥇